addons/hr_holidays/models/hr.py

```diff
@@ -0,0 +1,216 @@
+
+        for employee in self:
+            employee.leave_date_from = leave_data.get(employee.id, {}).get('leave_date_from')
+            employee.leave_date_to = leave_data.get(employee.id, {}).get('leave_date_to')
+            employee.current_leave_state = leave_data.get(employee.id, {}).get('current_leave_state')
+            employee.current_leave_id = leave_data.get(employee.id, {}).get('current_leave_id')
+
+    @api.multi
+    def _compute_leaves_count(self):
+        leaves = self.env['hr.holidays'].read_group([
+            ('employee_id', 'in', self.ids),
+            ('holiday_status_id.limit', '=', False),
+            ('state', '=', 'validate')
+        ], fields=['number_of_days', 'employee_id'], groupby=['employee_id'])
+        mapping = dict([(leave['employee_id'][0], leave['number_of_days']) for leave in leaves])
+        for employee in self:
+            employee.leaves_count = mapping.get(employee.id)
+
+    @api.multi
+    def _compute_show_leaves(self):
+        show_leaves = self.env['res.users'].has_group('base.group_hr_user')
+        for employee in self:
+            if show_leaves or employee.user_id == self.env.user:
+                employee.show_leaves = True
+            else:
+                employee.show_leaves = False
+
+    @api.multi
+    def _compute_absent_employee(self):
+        today_date = datetime.utcnow().date()
+        today_start = fields.Datetime.to_string(today_date)  # get the midnight of the current utc day
+        today_end = fields.Datetime.to_string(today_date + relativedelta(hours=23, minutes=59, seconds=59))
+        data = self.env['hr.holidays'].read_group([
+            ('employee_id', 'in', self.ids),
+            ('state', 'not in', ['cancel', 'refuse']),
+            ('date_from', '<=', today_end),
+            ('date_to', '>=', today_start),
+            ('type', '=', 'remove')
+        ], ['employee_id'], ['employee_id'])
+        result = dict.fromkeys(self.ids, False)
+        for item in data:
+            if item['employee_id_count'] >= 1:
+                result[item['employee_id'][0]] = True
+        for employee in self:
+            employee.is_absent_totay = result.get(employee.id)
+
+    @api.multi
+    def _search_absent_employee(self, operator, value):
+        today_date = datetime.utcnow().date()
+        today_start = fields.Datetime.to_string(today_date)  # get the midnight of the current utc day
+        today_end = fields.Datetime.to_string(today_date + relativedelta(hours=23, minutes=59, seconds=59))
+        holiday_values = self.env['hr.holidays'].search_read([
+            ('state', 'not in', ['cancel', 'refuse']),
+            ('date_from', '<=', today_end),
+            ('date_to', '>=', today_start),
+            ('type', '=', 'remove')
+        ], ['employee_id'])
+        absent_employee_ids = [holiday['employee_id'][0] for holiday in holiday_values if holiday['employee_id']]
```

Comment:
why not just search for holidays with employee_id != False, then just use those ids?